internal/utils/commit_stats.go

```diff
@@ -0,0 +1,366 @@
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/hhatto/gocloc"
+	"gopkg.in/src-d/enry.v1"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/go-git.v4/utils/binary"
+	"gopkg.in/src-d/go-git.v4/utils/ioutil"
+	"gopkg.in/src-d/go-git.v4/utils/merkletrie"
+)
+
+// LineKind defines the kind of a line in a file.
+type LineKind int
+
+const (
+	// Code represents a line of code.
+	Code LineKind = iota + 1
+	// Comment represents a line of comment.
+	Comment
+	// Blank represents an empty line.
+	Blank
+	// Other represents a line from any other kind.
+	Other
+)
+
+// CalculateCommitStats calculates the CommitStats for from commit to another.
+// if from is nil the first parent is used, if the commit is orphan the stats
+// are compared against a empty commit.
+func CalculateCommitStats(r *git.Repository, from, to *object.Commit) (*CommitStats, error) {
```

Comment:
If it's moved to `commitstats` package, I'd rename this to `Calculate` to avoid redundancy in `commitstats.CalculateCommitStats`.